bigquery/google/cloud/bigquery/job.py

```diff
@@ -572,12 +588,20 @@ def schema(self, value):
         :type value: list of :class:`SchemaField`
         :param value: fields describing the schema
 
-        :raises: TypeError if 'value' is not a sequence, or ValueError if
-                 any item in the sequence is not a SchemaField
+        :raises TypeError: If ``value`is not a sequence.
+        :raises ValueError: If any item in the sequence is not
+                            a ``SchemaField``.
         """
-        if not all(isinstance(field, SchemaField) for field in value):
-            raise ValueError('Schema items must be fields')
-        self._schema = tuple(value)
+        if not value:
+            self._schema = ()
+        else:
+            if not all(isinstance(field, SchemaField) for field in value):
+                raise ValueError('Schema items must be fields')
+            if self._configuration is not None and self.autodetect:
```

Comment:
@WillianFuks This used to say `if getattr(self, '_configuration', None) and self.autodetect`, but that was because in the constructor

```python
self.schema = schema
```

came before

```python
self._configuration = _LoadConfiguration()
```

So `self._configuration` would've been an `AttributeError`.

----

TL;DR: 

- Why did you check for `self._configuration` being not-`None`?
- Will it ever actually be `None`?